### DISCUSSION TITLE
Mojo / inference engine vs Torch compile

### DISCUSSION THREAD

**Discussion by timmermansjoy** [modular/modular#176] [Q&A] [answered]
in the [performance dashboard](https://performance.modular.com) it just says pytorch. However, Pytorch 2 also lets you compile your models. looking at the results in the image, they are about the same as from modular inference engine. could someone explain why you would use the modular inference engine over compiling your code with torch.compile? also can the inference engine do stuff like quantisation?

![](https://global-uploads.webflow.com/6392f64685456376914037d5/63a89feed2940d18ba818581_1670173813498-p-800.jpeg)

**Comment by iamtimdavis** [comment, marked as answer]
1. It’s unclear what hardware architecture or model you are comparing on PT 2.0 in the diagram above
2. All our numbers are on CPU (our GPU numbers will come and are incredible)
3. We outperform PyTorch across a large range of hardware (Intel, AMD, ARM etc) vs. the handpicked one illustrated here that isn't defined (see https://www.modular.com/engine#performance or swap around the Instance Types on the performance dashboard)
4. It’s a unified engine that enables multi-framework support - many users aren’t just using PyTorch (TensorFlow, JAX etc)
5. It integrates natively with Mojo🔥 for a completely new high performance programming model that enables many things outside of just pure model execution performance
6. We support quantization and it will support many other HW types like edge deployments
7. There are so many other aspects that we will be talking about soon that gives it direct advantages 🚀

**Reply by dogterbox** [reply]
Do you have an update ?
I want to benchmark Mojo on CPU vs Pytorch on GPU.
